holoviews/core/util.py

```diff
@@ -692,18 +704,69 @@ def isnumeric(val):
         return False
 
 
+def asarray(arraylike, strict=True):
+    """
+    Converts arraylike objects to NumPy ndarray types. Errors if
+    object is not arraylike and strict option is enabled.
+    """
+    if isinstance(arraylike, np.ndarray):
+        return arraylike
+    elif isinstance(arraylike, list):
+        return np.asarray(arraylike, dtype=object)
+    elif not isinstance(arraylike, np.ndarray) and isinstance(arraylike, arraylike_types):
+        return arraylike.values
+    elif hasattr(arraylike, '__array__'):
+        return np.asarray(arraylike)
+    elif strict:
+        raise ValueError('Could not convert %s type to array' % type(arraylike))
+    return arraylike
+
+
+nat_as_integer = np.datetime64('NAT').view('i8')
+
+def isnat(val):
+    """
+    Checks if the value is a NaT. Should only be called on datetimelike objects.
+    """
+    if isinstance(val, (np.datetime64, np.timedelta64)):
+        if numpy_version >= '1.13':
+            return np.isnat(val)
+        else:
+            return val.view('i8') == nat_as_integer
+    elif pd and val is pd.NaT:
```

Comment:
Has pd.NaT always been available, or only since a certain Pandas release?  If the latter, is that Pandas release a minimum requirement for HoloViews?

Reply:
I see StackOverFlow issues about pd.NaT going back to 2014, so that doesn't seem a major worry.

Reply:
Ok.

Reply:
That said we should make a list of optional dependencies and determine minimum versions for them.